x-pack/legacy/plugins/maps/public/layers/sources/client_file_source/geojson_file_source.js

```diff
@@ -125,6 +127,22 @@ export class GeojsonFileSource extends AbstractVectorSource {
     );
   }
 
+  async getGeoJsonWithMeta() {
```

Comment:
I'd add a safety check here to check on the existence of `__featureCollection`. This source _could_ be created by a client who's passing in a descriptor without the `__featureCollection` property, especially now we have embeddables (e.g. consider an embeddable-client just injecting data with a GeoJsonSource and misconfigurin the descriptor). We don't want this call to throw.

You could also enforce that there's always an empty featureCollection for the `__featureCollection` property in the `GeoJsonFileSource#constructor` .

Reply:
> You could also enforce that there's always an empty featureCollection for the __featureCollection property in the GeoJsonFileSource#constructor .

I like that. Done!